crates/red_knot_python_semantic/src/lint.rs

```diff
@@ -0,0 +1,205 @@
+use itertools::Itertools;
+use ruff_db::diagnostic::{LintName, Severity};
+
+#[derive(Debug, Clone)]
+pub struct LintMetadata {
+    /// The unique identifier for the lint.
+    pub name: LintName,
+
+    /// A one-sentence summary of what the lint catches.
+    pub summary: &'static str,
+
+    /// An in depth explanation of the lint in markdown. Covers what the lint does, why it's bad and possible fixes.
+    ///
+    /// The documentation may require post-processing to be rendered correctly. For example, lines
+    /// might have leading or trailing whitespace that should be removed.
+    pub raw_documentation: &'static str,
+
+    /// The default level of the lint if the user doesn't specify one.
+    pub default_level: Level,
+
+    pub status: LintStatus,
+
+    /// Location where this lint is declared: `file_name:line`
+    pub source: &'static str,
```

Comment:
Do we want this to be part of a release binary? Having potentially hundreds of `"crates/red_knot_python_semantic/src/types/diagnostic.rs:123"` strings in the `.rodata` section seems like a bit of a waste?

Reply:
This is as good point. I consider it fairly small compared to e.g. the documentation of every rule and all vendored stubs but I don't mind gating it behind a feature that we only activate when generating the reference documentation if you think it's worth doing

Reply:
>  consider it fairly small compared to e.g. the documentation of every rule

Right. For 1000 lints, it would be ~ 100 KiB, so feel free to ignore it :smile:

Reply:
I could change it to two fields: `file` and `line` to possibly safe some space for the `:` and line number :laughing:

Reply:
I'd also expect LTO to remove the data if it isn't referenced anywhere... But I haven't verified if LTO is capable of doing this.